docs/core/diagnostics/debug-stackoverflow.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Debugging StackOverflow errors
+description: Learn how to diagnose StackOverflow exceptions
+ms.topic: tutorial
+ms.date: 12/22/2020
+---
+
+# Debugging StackOverflow errors
+
+A <xref:System.StackOverflowException> is thrown when when the execution stack overflows because it contains too many nested method calls.  
+
+For example, suppose you have an app as follows:
+
+````
+using System;
+
+namespace temp
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Main(args); // oops this recursion won't stop
+        }
+    }
+}
+````
+
+The Main method will continuously call itself until there is no more stack space.  Once there is no more stack space, execution cannot continue and so it will throw a <xref:System.StackOverflowException>.  
+
+````
+> dotnet run
+Stack overflow.
+````
+
+> [!NOTE]
+On .NET 5 and later, the callstack is output to the console.
+
+> [!NOTE]
+This article describes how to debug a StackOverflow with lldb. If you are running on Windows, we suggest debugging the app with [Visual Studio](/visualstudio/debugger/what-is-debugging) or [Visual Studio Code](https://code.visualstudio.com/Docs/editor/debugging).  
+
+## Example
+
+1. Run the app with it configured to collect a dump on crash
+
+    ````
+    > export COMPlus_DbgEnableMiniDump=1
+    > dotnet run
+    Stack overflow.
+    Writing minidump with heap to file /tmp/coredump.6412
+    Written 58191872 bytes (14207 pages) to core file
+    ````
+
+2. Debug the dump in lldb to see the failing stack
```

Comment:
Before we open it up with lldb, should we add a step here that tells them to install SOS with a link to dotnet-sos?

Reply:
Great idea!  I've pushed an update for this.